### ISSUE TITLE
Training on Google Colab

### ISSUE THREAD

**Issue by TBauer2000** [ika-rwth-aachen/Cam2BEV#36]
Hi,

I'm trying to train your UNetXST model for the front view only on a Google Colab notebook, but it takes forever for each epoch. (2hrs +)
The notebook is running Python 3.10.11 with CUDA 11.8 and Tensorflow 2.12.0 preinstalled. In requirements.txt you suggest to train with Tensorflow<2.5.0, but this seems to affect only the Deeplab models.

Another thing that I have noticed is the TF-TRT Warning: Could not find TensorRT... do I need to install TensorRT for training?

Thank you in advance!

Below is the output of the terminal while training:

2023-05-22 12:35:07.066135: I tensorflow/core/util/port.cc:110] oneDNN custom operations are on. You may see slightly different numerical results due to floating-point round-off errors from different computation orders. To turn them off, set the environment variable `TF_ENABLE_ONEDNN_OPTS=0`.
2023-05-22 12:35:07.119840: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F AVX512_VNNI FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-05-22 12:35:08.005860: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
Found 32190 training samples
Found 3172 validation samples
2023-05-22 12:36:46.183583: W tensorflow/core/common_runtime/gpu/gpu_bfc_allocator.cc:47] Overriding orig_value setting because the TF_FORCE_GPU_ALLOW_GROWTH environment variable is set. Original config value was 0.
2023-05-22 12:36:46.183643: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1635] Created device /job:localhost/replica:0/task:0/device:GPU:0 with 38284 MB memory:  -> device: 0, name: NVIDIA A100-SXM4-40GB, pci bus id: 0000:00:04.0, compute capability: 8.0
Built data pipeline for training
Built data pipeline for validation
Compiled model uNetXST.py
Starting training...
2023-05-22 12:36:59.890772: I tensorflow/core/common_runtime/executor.cc:1197] [/device:CPU:0] (DEBUG INFO) Executor start aborting (this does not indicate an error and you can ignore this message): INVALID_ARGUMENT: You must feed a value for placeholder tensor 'Placeholder/_1' with dtype string and shape [32190]
	 [[{{node Placeholder/_1}}]]
2023-05-22 12:36:59.891129: I tensorflow/core/common_runtime/executor.cc:1197] [/device:CPU:0] (DEBUG INFO) Executor start aborting (this does not indicate an error and you can ignore this message): INVALID_ARGUMENT: You must feed a value for placeholder tensor 'Placeholder/_1' with dtype string and shape [32190]
	 [[{{node Placeholder/_1}}]]
Epoch 1/100
2023-05-22 12:37:19.169642: E tensorflow/core/grappler/optimizers/meta_optimizer.cc:954] layout failed: INVALID_ARGUMENT: Size of values 0 does not match size of permutation 4 @ fanin shape inmodel_5/dropout/dropout/SelectV2-2-TransposeNHWCToNCHW-LayoutOptimizer
2023-05-22 12:37:25.133130: I tensorflow/compiler/xla/stream_executor/cuda/cuda_dnn.cc:424] Loaded cuDNN version 8700
2023-05-22 12:37:31.275985: I tensorflow/compiler/xla/service/service.cc:169] XLA service 0x7f0603f4c350 initialized for platform CUDA (this does not guarantee that XLA will be used). Devices:
2023-05-22 12:37:31.276032: I tensorflow/compiler/xla/service/service.cc:177]   StreamExecutor device (0): NVIDIA A100-SXM4-40GB, Compute Capability 8.0
2023-05-22 12:37:31.420157: I tensorflow/compiler/mlir/tensorflow/utils/dump_mlir_util.cc:269] disabling MLIR crash reproducer, set env var `MLIR_CRASH_REPRODUCER_DIRECTORY` to enable.
2023-05-22 12:37:31.919526: I ./tensorflow/compiler/jit/device_compiler.h:180] Compiled cluster using XLA!  This line is logged at most once for the lifetime of the process.
  54/6438 [..............................] - ETA: 2:13:38 - loss: 1.3788 - categorical_accuracy: 0.5008 - mean_io_u_with_one_hot_labels: 0.2887

**Comment by lreiher** [comment]
I have not tested Google Colab, but can train uNetXST for one epoch in about 10min on the 2_F dataset on our private NVIDIA A100, see log below. This is with the recommended Python environment.
- Python 3.7
- TensorFlow 2.5 (and everything else from `requirements.txt`
- CUDA 11.2
- cuDNN 8.1

Please note that we are not actively upgrading this repository to more recent Python / TensorFlow / CUDA versions anymore.

```
023-05-24 08:36:24.910143: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
Found 32246 training samples
Found 3172 validation samples
2023-05-24 08:36:26.031596: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcuda.so.1
2023-05-24 08:36:26.113491: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: 
pciBusID: 0000:41:00.0 name: NVIDIA A100-SXM4-40GB computeCapability: 8.0
coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.59GiB deviceMemoryBandwidth: 1.41TiB/s
2023-05-24 08:36:26.113542: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
2023-05-24 08:36:26.115711: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11
2023-05-24 08:36:26.115751: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11
2023-05-24 08:36:26.116522: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcufft.so.10
2023-05-24 08:36:26.116698: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcurand.so.10
2023-05-24 08:36:26.117295: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusolver.so.11
2023-05-24 08:36:26.117822: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcusparse.so.11
2023-05-24 08:36:26.117932: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8
2023-05-24 08:36:26.121095: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0
2023-05-24 08:36:26.121608: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-05-24 08:36:26.152714: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1733] Found device 0 with properties: 
pciBusID: 0000:41:00.0 name: NVIDIA A100-SXM4-40GB computeCapability: 8.0
coreClock: 1.41GHz coreCount: 108 deviceMemorySize: 39.59GiB deviceMemoryBandwidth: 1.41TiB/s
2023-05-24 08:36:26.156884: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1871] Adding visible gpu devices: 0
2023-05-24 08:36:26.156937: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
2023-05-24 08:36:27.210685: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1258] Device interconnect StreamExecutor with strength 1 edge matrix:
2023-05-24 08:36:27.210760: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1264]      0 
2023-05-24 08:36:27.210769: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1277] 0:   N 
2023-05-24 08:36:27.216527: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1418] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 37774 MB memory) -> physical GPU (device: 0, name: NVIDIA A100-SXM4-40GB, pci bus id: 0000:41:00.0, compute capability: 8.0)
Built data pipeline for training
Built data pipeline for validation
Compiled model uNetXST.py
Starting training...
2023-05-24 08:36:34.683173: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:176] None of the MLIR Optimization Passes are enabled (registered 2)
2023-05-24 08:36:34.711923: I tensorflow/core/platform/profile_utils/cpu_utils.cc:114] CPU Frequency: 2250035000 Hz
Epoch 1/100
2023-05-24 08:36:46.501740: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublas.so.11
2023-05-24 08:36:48.682107: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcublasLt.so.11
2023-05-24 08:36:48.683088: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudnn.so.8
2023-05-24 08:36:50.143441: I tensorflow/stream_executor/cuda/cuda_dnn.cc:359] Loaded cuDNN version 8100
6449/6449 [==============================] - ETA: 0s - loss: 0.2305 - categorical_accuracy: 0.9356 - mean_io_u_with_one_hot_labels: 0.75832023-05-24 08:43:53.638021: I tensorflow/stream_executor/cuda/cuda_blas.cc:1838] TensorFloat-32 will be used for the matrix multiplication. This will only be logged once.
6449/6449 [==============================] - 657s 99ms/step - loss: 0.2305 - categorical_accuracy: 0.9356 - mean_io_u_with_one_hot_labels: 0.7583 - val_loss: 0.1702 - val_categorical_accuracy: 0.9477 - val_mean_io_u_with_one_hot_labels: 0.7897
Epoch 2/100
6449/6449 [==============================] - 619s 96ms/step - loss: 0.1468 - categorical_accuracy: 0.9544 - mean_io_u_with_one_hot_labels: 0.8116 - val_loss: 0.1497 - val_categorical_accuracy: 0.9535 - val_mean_io_u_with_one_hot_labels: 0.8091
```

**Comment by TBauer2000** [comment]
Alright thank you for your quick response! Seems like I have to get your requirements running on a Colab Notebook :)
